src/main/java/teammates/storage/sqlentity/Notification.java

```diff
@@ -0,0 +1,301 @@
+package teammates.storage.sqlentity;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+import org.hibernate.annotations.CreationTimestamp;
+import org.hibernate.annotations.UpdateTimestamp;
+
+import teammates.common.datatransfer.NotificationStyle;
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.util.FieldValidator;
+import teammates.common.util.JsonUtils;
+import teammates.common.util.SanitizationHelper;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+
+/**
+ * Represents a unique notification in the system.
+ */
+@Entity
+@Table(name = "Notifications")
+public class Notification extends BaseEntity {
+
+    @Id
+    private String notificationId;
+
+    @Column(nullable = false)
+    private Instant startTime;
+
+    @Column(nullable = false)
+    private Instant endTime;
+
+    @Column(nullable = false)
+    @Enumerated(EnumType.ORDINAL)
+    private NotificationStyle style;
+
+    @Column(nullable = false)
+    @Enumerated(EnumType.ORDINAL)
+    private NotificationTargetUser targetUser;
+
+    @Column(nullable = false)
+    private String title;
+
+    @Column(nullable = false)
+    private String message;
+
+    @Column(nullable = false)
+    private boolean shown;
+
+    @CreationTimestamp
+    @Column(nullable = false, updatable = false)
+    private Instant createdAt;
+
+    @UpdateTimestamp
+    @Column
+    private Instant updatedAt;
+
+    /**
+     * Instantiates a new notification, with ID randomly generated and time fields filled automatically.
+     *
+     * @param startTime start time for the notification to be shown to users
+     * @param endTime notifications are hidden from users after endTime
+     * @param style style of the notification (e.g. success, warning, etc.)
+     * @param targetUser student or instructor
+     * @param title title of the notification
+     * @param message message body of the notification
+     */
+    public Notification(Instant startTime, Instant endTime, NotificationStyle style, NotificationTargetUser targetUser,
```

Comment:
we might not need this constructor if we have the builder

Reply:
Indeed. This constructor is not used anywhere currently.